File: backend/app/schemas/wallet_address.py
from uuid import UUID
from pydantic import BaseModel, ConfigDict
class WalletAddressCreate(BaseModel):
    asset_id: UUID
    network: str
    address: str
    address_type: str = "DEPOSIT"
class WalletAddressResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    wallet_id: UUID
    asset_id: UUID
    network: str
    address: str
    address_type: str
    status: str

class WalletAddressGenerateRequest(BaseModel):
    asset: str
    network: str

class WalletAddressGenerateResponse(BaseModel):
    asset: str
    network: str
    address: str
    derivation_index: int
    newly_generated: bool    

File: mobile/lib/models/wallet_address.dart
class WalletAddress {
  const WalletAddress({
    required this.id,
    required this.walletId,
    required this.assetId,
    required this.network,
    required this.address,
    required this.addressType,
    required this.status,
  });

  final String id;
  final String walletId;
  final String assetId;
  final String network;
  final String address;
  final String addressType;
  final String status;

  factory WalletAddress.fromJson(Map<String, dynamic> json) => WalletAddress(
        id: json['id'].toString(),
        walletId: json['wallet_id'].toString(),
        assetId: json['asset_id'].toString(),
        network: json['network']?.toString() ?? '',
        address: json['address']?.toString() ?? '',
        addressType: json['address_type']?.toString() ?? '',
        status: json['status']?.toString() ?? '',
      );
}
